tests/scripts/all.sh

```diff
@@ -676,15 +676,19 @@ if uname -a | grep -F Linux >/dev/null; then
 fi
 
 if uname -a | grep -F x86_64 >/dev/null; then
-    msg "build: i386, make, gcc" # ~ 30s
+    msg "build: i386, make, gcc (ASan build)" # ~ 30s
     cleanup
-    make CC=gcc CFLAGS='-Werror -Wall -Wextra -m32'
+    cp "$CONFIG_H" "$CONFIG_BAK"
+    scripts/config.pl full
+    make CC=gcc CFLAGS='-Werror -Wall -Wextra -m32 -fsanitize=address'
```

Comment:
This builds with `-O0`, meaning we might not get full warnings, but more importantly we're not testing the assembly code. I think we want to have two builds and tests: one with `-O0` and the other with `-O1` (with comments explaining why we need the two).

Reply:
Have added an additional test as suggested.